AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.kt

```
@@ -415,6 +411,23 @@ open class AnkiDroidApp : Application() {
             }
         }
 
+        /** Load the libraries to allow access to Anki-Android-Backend */
+        @NeedsTest("Not calling this in the ContentProvider should have failed a test")
+        fun makeBackendUsable(context: Context) {
+            // Robolectric uses RustBackendLoader.ensureSetup()
+            if (Build.FINGERPRINT == "robolectric") return
+
+            try {
+                // Prevent sqlite throwing error 6410 due to the lack of /tmp on Android
+                Os.setenv("TMPDIR", context.cacheDir.path, false)
+                // Load backend library
+                System.loadLibrary("rsdroid")
+            } catch (e: Error) {
+                // this routine /may/ be called more than once: we'll know if/when it occurs
```

Just to clarify: setenv and loadLibrary() are idempotent IIRC, so this would only throw if some other error occurs like the library not being able to be loaded. Not sure if logging/swallowing the error is the best option in that case, since the app can't be effectively used afterwards anyway.